### ISSUE TITLE
Improvements in Sum.doit() for functions having the +/- operator .

### ISSUE THREAD

**Issue by anutosh491** [sympy/sympy#22178]
While trying to approach another issue(#14410) , I realized that Sum.doit() doesn't work as expected and certainly needs improvements to solve a couple of issues I browsed through .

1) `Sum.doit()` for functions having the  +/-  operator doesn't always give expected outputs, for eg
```
>>> Sum((3+x) , (x,1,oo)).doit()
oo
>>> Sum((3-x) , (x,1,oo)).doit()
Sum(3 - x, (x, 1, oo))
>>> Sum((3-x) , (x,4,oo)).doit()
-oo
```
At first look this indicates  that maybe `doit`  is checking for the first term whether +ve/-ve and then returns `oo/-oo` accordingly .But the next example shows that might not be the case 

```
>>> Sum( x**2 - x**3, (x,1,oo)).doit()
-oo
>>> Sum( x**2 - x**3, (x,0,oo)).doit()
Sum(-x**3 + x**2, (x, 0, oo))
>>> Sum( x**2 - x**3, (x,-1,oo)).doit()
Sum(-x**3 + x**2, (x, -1, oo))
```

This example shows that if there's a possibility to get both +ve and -ve terms from the Sum ,then `Sum.doit()` doesn't perform as expected and the first intuition is probably not correct .

2) An alternating series passed inside Sum either with or without  the +/- operators  eg `Sum(((-1)**x *(x**2 + 1)/x) , (x,1,oo))`  or  `Sum((3 + (-1)**x *(x**2 + 1)/x) , (x,1,oo))`  which would eventually diverge should be returning `None `cause this would oscillating between +oo and -oo eventually . As of now this is yet to be implemented
```
>>> Sum(((-1)**x *(x**2 + 1)/x) , (x,1,oo)).doit()
Sum((-1)**x*(x**2 + 1)/x, (x, 1, oo))
>>> Sum((3 + (-1)**x *(x**2 + 1)/x) , (x,1,oo)).doit()
Sum((-1)**x*(x**2 + 1)/x + 3, (x, 1, oo))
```

**Comment by anutosh491** [comment]
I'll be going through how doit works for Sum and try to figure out a way to fix this . I have couple ideas in mind .Any suggestions/thoughts are welcome . Thank you !

**Comment by oscargus** [comment]
I think it checks for non-negative or non-positive (so including 0), e.g. `Sum((3-x) , (x,3,oo)).doit()` is `-oo`. So maybe a way is to check if one can separate the summation into two: one with positive and one with negative elements. If one is finite and the other is infinite, it will be infinite (if both are finite => finite, both infinite => cannot tell). I guess the problem is that it is quite easy to do by "looking" at the summation and maybe not so easy otherwise... Finding zeros for the summation argument and looking at derivatives may be one way to possibly identify this.

**Comment by anutosh491** [comment]
Yeah I'm trying to implement a similar logic , this would be really useful as couple of issues labelled concrete need would need this to be done first . I have been trying since a day or two . The problem is that `Sum.doit() `deals with such summations having +/- operator by splitting them into parts and getting result seperately which is not wrong at all but in such cases ,3 would return `oo` and something like -sqrt(x) of x would return `-oo ` ,hence oo - oo gives `nan` rather than considering the above logic you said .Hence Sum(3-x , (x, 1, oo)) doesn't work properly while other 2 do !
